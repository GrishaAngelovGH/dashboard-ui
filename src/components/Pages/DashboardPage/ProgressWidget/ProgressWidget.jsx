import ProgressBar from 'react-bootstrap/ProgressBar'
import './ProgressWidget.scss'

const ProgressWidget = ({ title, subtitle, value, progress, label }) => (
	<div className='bg-white rounded progress-widget p-2'>
		<div className='row'>
			<div className='col-6 col-md-6'>
				<h5>{title}</h5>
				<p className='content-color'>{subtitle}</p>
			</div>
			<div className='col-6 col-md-6 d-flex justify-content-end'>
				<div className='content-value'>{value}</div>
			</div>
		</div>
		<div className='row'>
			<div className='col-md-12'>
				<ProgressBar animated now={progress} />
			</div>
		</div>

		<div className='d-flex justify-content-between content-color'>
			<div>{label}</div>
			<div>100%</div>
		</div>
	</div>
)

export default ProgressWidget