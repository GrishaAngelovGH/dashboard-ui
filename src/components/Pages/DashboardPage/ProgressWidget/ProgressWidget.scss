.progress-widget {
	width: 350px;
	height: 150px;
	box-shadow: 0px 0px 10px 3px #ccc;
}

.content-value {
	color: orange;
	font-size: 30px;
	font-weight: bold;
}

.content-color {
	color: orange;
}
