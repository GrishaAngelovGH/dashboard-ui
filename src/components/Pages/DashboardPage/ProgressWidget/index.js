import ProgressWidget from './ProgressWidget'

export default ProgressWidget