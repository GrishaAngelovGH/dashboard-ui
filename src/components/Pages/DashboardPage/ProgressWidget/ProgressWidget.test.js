import { render } from '@testing-library/react'

import ProgressWidget from './ProgressWidget'

test('should render component', () => {
    const container = render(
        <ProgressWidget
            title={'Total Orders'}
            subtitle={'Last year expenses'}
            value={'$3M'}
            progress={65}
            label={'YoY Growth'}
        />
    )

    expect(container).toMatchSnapshot()
})