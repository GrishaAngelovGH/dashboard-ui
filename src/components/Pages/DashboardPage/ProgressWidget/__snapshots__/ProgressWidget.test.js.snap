// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProgressWidget component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-white rounded progress-widget p-2"
      >
        <div
          class="row"
        >
          <div
            class="col-6 col-md-6"
          >
            <h5>
              Total Orders
            </h5>
            <p
              class="content-color"
            >
              Last year expenses
            </p>
          </div>
          <div
            class="col-6 col-md-6 d-flex justify-content-end"
          >
            <div
              class="content-value"
            >
              $3M
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="progress"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="65"
                class="progress-bar progress-bar-animated progress-bar-striped"
                role="progressbar"
                style="width: 65%;"
              />
            </div>
          </div>
        </div>
        <div
          class="d-flex justify-content-between content-color"
        >
          <div>
            YoY Growth
          </div>
          <div>
            100%
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-white rounded progress-widget p-2"
    >
      <div
        class="row"
      >
        <div
          class="col-6 col-md-6"
        >
          <h5>
            Total Orders
          </h5>
          <p
            class="content-color"
          >
            Last year expenses
          </p>
        </div>
        <div
          class="col-6 col-md-6 d-flex justify-content-end"
        >
          <div
            class="content-value"
          >
            $3M
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="progress"
          >
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="65"
              class="progress-bar progress-bar-animated progress-bar-striped"
              role="progressbar"
              style="width: 65%;"
            />
          </div>
        </div>
      </div>
      <div
        class="d-flex justify-content-between content-color"
      >
        <div>
          YoY Growth
        </div>
        <div>
          100%
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
